Assignment4/Assignment4.py

```diff
@@ -0,0 +1,246 @@
+
+    return data, X, Y
+
+
+def Cat_resp2():
+    data = pd.read_csv("/content/drive/MyDrive/heart.csv")
+
+    # Separating the data to target and response variables
+    X = data[["age", "sex", "cp", "trtbps", "chol", "fbs", "restecg", "thalachh", "exng", "oldpeak", "slp", "caa",
+              "thall"]]  # PREDICTOR
+    Y = data[["output"]]  # RESPONSE
+
+    return data, X, Y
+
+
+# Calling the Continous response function
+
+data, X, Y = Cat_resp2()
+
+# Creating a table to store the results
+Summary_Table = pd.DataFrame()
+
+# initiating the lists to store the data in a list wrt
+Response, Predictor, Plot, p_value, t_score, RF_VarImp, MWR_Unweighted, MWR_Weighted = [], [], [], [], [], [], [], []
+
+# calculatin the random forest variable importance
+model = RandomForestRegressor()
+model.fit(X, Y)
+
+
+# continuous predictor vs continuous response
+def cat_response_cat_predictor():
+    x = numpy.array(X.iloc[:, i])
+    y = numpy.array(Y)
+
+    X_2 = [1 if abs(x_) > 0.5 else 0 for x_ in x]
+    Y_2 = [1 if abs(y_) > 0.5 else 0 for y_ in y]
+
+    conf_matrix = confusion_matrix(X_2, Y_2)
+
+    fig = go.Figure(
+        data=go.Heatmap(z=conf_matrix, zmin=0, zmax=conf_matrix.max())
+    )
+    fig.update_layout(
+        title="Categorical Predictor by Categorical Response (with relationship)",
+        xaxis_title="Response",
+        yaxis_title="Predictor",
+    )
+    fig.show()
+
+    return
+
+
+# Cat response continuous predictor
+
+def cat_resp_cont_predictor():
+    fig = px.histogram(data, x=X.columns[i], color=Y.columns[0], marginal="rug", hover_data=X.columns)
+
+    fig.update_layout(
+        title="Continuous Predictor by Categorical Response",
+        xaxis_title="Predictor",
+        yaxis_title="Distribution",
+    )
+    fig.show()
+    return
+
+
+# Cont Response Cat Predictor
+
+def cont_resp_cat_predictor():
+    fig = px.violin(data, y=Y.columns[0], x=X.columns[i], box=True, hover_data=X.columns)
+    fig.show()
+    return
+
+
+# Cont Response Cont Predictor
+
+def cont_response_cont_predictor():
+    x = X.iloc[:, i]
+    y = Y.iloc[:, 0]
+
+    fig = px.scatter(x=x, y=y, trendline="ols")
+    fig.update_layout(
+        title="Continuous Response by Continuous Predictor",
+        xaxis_title="Predictor",
+        yaxis_title="Response",
+    )
+    fig.show()
+
+    return
+
+
+# looping the columns to verify the data type
+
+P_Columns = list(X.columns)
+R_columns = list(Y.columns)
+Y1 = Y.iloc[:, 0]
+
+if len(pd.unique(Y1)) <= 2:
+    for i in range(0, X.shape[1]):
+        Response.append(R_columns[0])
+        n = len(pd.unique(X.iloc[:, i]))
+
+        if n <= 4:
+
+            # appending the column name along with its type
+            Predictor.append(P_Columns[i] + '(Categorical)')
+
+            # calling the plot for this type
+            cat_response_cat_predictor()
+
+            t_score.append("NA")
+            Plot.append("NA")
+            p_value.append("NA")
```

Comment:
For calculating random forest scores, you can create a new data frame with continuous predictors columns.